toucan_connectors/mongo/mongo_connector.py

```diff
@@ -220,6 +221,25 @@ def get_slice(
             total_count = len(df)
         return DataSlice(df, total_count)
 
+    def get_df_with_regex(
+        self,
+        data_source: MongoDataSource,
+        field: str,
+        regex,
+        permissions: Optional[str] = None,
+        limit: Optional[int] = None,
+    ) -> pd.DataFrame:
+        # Create a copy in order to keep the original (deepcopy-like)
+        data_source = MongoDataSource.parse_obj(data_source)
+        data_source.query = normalize_query(data_source.query, data_source.parameters)
+        data_source.query[0]['$match'] = {
+            '$and': [data_source.query[0]['$match']]
+            + [{field: {'$regex': Regex.from_native(regex)}}]
+        }
+        if limit is not None:
+            data_source.query.append({'$limit': limit})
+        return self.get_df(data_source, permissions)
```

Comment:
It works but I'm afraid we are putting the same logic at different places. I feel like what you did is almost the same as `get_slice` but instead of permissions we add the regex filter.
Wouldn't it be possible to use `get_slice` ?

Reply:
As it may be too long to do (and out of scope), I'll probably open an issue if the PR is merged as such.

Reply:
No that's a good idea! I'll do this